pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -584,3 +593,163 @@ func (a *Actuator) nodeAddresses(host *bmh.BareMetalHost) ([]corev1.NodeAddress,
+	}
+
+	baremetalhost.Annotations[rebootAnnotation] = ""
+
+	err := a.client.Update(ctx, baremetalhost)
+	if err != nil {
+		log.Printf("failed to add reboot annotation to %s: %s", baremetalhost.Name, err.Error())
+	}
+
+	return err
+}
+
+//requestPowerOn adds reboot annotation on baremetalhost which signal BMO to power off the machine
+func (a *Actuator) requestPowerOn(ctx context.Context, baremetalhost *bmh.BareMetalHost) error {
+	if baremetalhost.Annotations == nil {
+		baremetalhost.Annotations = make(map[string]string)
+	}
+
+	if _, rebootPending := baremetalhost.Annotations[rebootAnnotation]; !rebootPending {
+		return &clustererror.RequeueAfterError{}
+	}
+
+	delete(baremetalhost.Annotations, rebootAnnotation)
+
+	err := a.client.Update(ctx, baremetalhost)
+	if err != nil {
+		log.Printf("failed to remove reboot annotation from %s: %s", baremetalhost.Name, err.Error())
+	}
+
+	return err
+}
+
+// deleteMachineNode deletes the node that mapped to specified machine
+func (a *Actuator) deleteNode(ctx context.Context, node *corev1.Node) error {
+	err := a.client.Delete(ctx, node)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return &clustererror.RequeueAfterError{}
+		}
+		log.Printf("Failed to delete node %s: %s", node.Name, err.Error())
+	}
+	return err
+}
+
+// getNodeByMachine returns the node object referenced by machine
+func (a *Actuator) getNodeByMachine(ctx context.Context, machine *machinev1.Machine) (*corev1.Node, error) {
+	if machine.Status.NodeRef == nil {
+		return nil, errors.NewNotFound(corev1.Resource("ObjectReference"), machine.Name)
+	}
+
+	node := &corev1.Node{}
+	key := client.ObjectKey{
+		Name:      machine.Status.NodeRef.Name,
+		Namespace: machine.Status.NodeRef.Namespace,
+	}
+
+	if err := a.client.Get(ctx, key, node); err != nil {
+		return nil, err
+	}
+	return node, nil
+}
+
+//remediateIfNeeded will try to remediate unhealthy nodes (annotated by MHC)
+func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1.Machine, baremetalhost *bmh.BareMetalHost) error {
+	if len(machine.Annotations) == 0 {
+		return nil
+	}
+
+	if _, needsRemediation := machine.Annotations[externalRemediationAnnotation]; !needsRemediation {
+		return nil
+	}
+
+	if _, remediationInProgress := machine.Annotations[remediationInProgressAnnotation]; !remediationInProgress {
+		if !hasRebootAnnotation(baremetalhost) {
+			log.Printf("Found an unhealthy machine, requesting power off. Machine name: %s", machine.Name)
+			return a.requestPowerOff(ctx, baremetalhost)
+		}
+
+		if baremetalhost.Status.PoweredOn {
+			return nil
+		}
+
+		//we need this annotation to differentiate between unhealthy machine that
+		//needs remediation to unhealthy machine that just got remediated
+		return a.addRemediationInProgressAnnotation(ctx, machine)
+	}
+
+	node, err := a.getNodeByMachine(ctx, machine)
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			log.Printf("failed to get Node from Machine %s: %s", machine.Name, err.Error())
```

Comment:
```suggestion
			log.Printf("Failed to get Node from Machine %s: %s", machine.Name, err.Error())
```

Most of your other logs start with a capital 🤷‍♂️ 😊